fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -35,37 +35,43 @@ class ExponentiatedGradient(BaseEstimator, MetaEstimatorMixin):
         violation is at most :code:`2*(eps+best_gap)`
     :type eps: float
 
-    :param T: Maximum number of iterations
-    :type T: int
+    :param max_iter: Maximum number of iterations
+    :type max_iter: int
 
     :param nu: Convergence threshold for the duality gap, corresponding to a
         conservative automatic setting based on the statistical uncertainty in measuring
         classification error
     :type nu: float
 
-    :param eta_mul: Initial setting of the learning rate
-    :type eta_mul: float
+    :param eta0: Initial setting of the learning rate
+    :type eta0: float
+
+    :param run_linprog_step: if True each step of exponentiated gradient is followed by the saddle
+        point optimization over the convex hull of classifiers returned so far; default True
+    :type run_linprog_step: bool
     """
 
-    def __init__(self, estimator, constraints, eps=0.01, T=50, nu=None, eta_mul=2.0):  # noqa: D103
-        self._estimator = estimator
-        self._constraints = constraints
-        self._eps = eps
-        self._T = T
-        self._nu = nu
-        self._eta_mul = eta_mul
-
-        self._best_gap = None
-        self._predictors = None
-        self._weights = None
-        self._last_t = None
-        self._best_t = None
-        self._n_oracle_calls = 0
-        self._n_oracle_calls_dummy_returned = 0
-        self._oracle_execution_times = None
-        self._lambda_vecs = pd.DataFrame()
-        self._lambda_vecs_LP = pd.DataFrame()
-        self._lambda_vecs_lagrangian = pd.DataFrame()
+    def __init__(self, estimator, constraints, eps=0.01, max_iter=50, nu=None,
+                 eta0=2.0, run_linprog_step=True):  # noqa: D103
+        self.estimator = estimator
+        self.constraints = constraints
+        self.eps = eps
+        self.max_iter = max_iter
+        self.nu = nu
+        self.eta0 = eta0
+        self.run_linprog_step = run_linprog_step
+
+        self.best_gap_ = None
```

Comment:
These initializations should not be here, but at the very beginning of fit (only if we need them). See: https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/utils/validation.py#L956

Reply:
Most of them are not even needed. We just didn't like the practice of suddenly defining things outside of `__init__` so far but that's simply how Python folks do that...

Reply:
We can possibly hold off with this for a future PR--your call.